elixir/grade-school/lib/school.ex

```diff
@@ -0,0 +1,38 @@
+defmodule School do
+  @moduledoc """
+  Simulate students in a school.
+
+  Each student is in a grade.
+  """
+
+  @doc """
+  Add a student to a particular grade in school.
+  """
+  @spec add(map, String.t(), integer) :: map
+
+  def add(db, name, grade) do
+    case Map.fetch(db, grade) do
+      :error -> db = Map.put(db, grade, [name])
+      {:ok, list} -> db = Map.put(db, grade, [name | list])
```

Comment:
Why are you assigning the variable `db` on these 2 lines?

This is likely causing a compiler warning.

Reply:
@alanvardy For some reason I thought that if I didn't do that my values wouldn't be kept in there.